src/components/AutoUpdateTime.js

```diff
@@ -0,0 +1,99 @@
+import {View} from 'react-native';
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+import DateUtils from '../libs/DateUtils';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import Timers from '../libs/Timers';
+import Text from './Text';
+
+const propTypes = {
+    /** Timezone of the user from their personal details */
+    timezone: PropTypes.shape({
+        /** Value of selected timezone */
+        selected: PropTypes.string,
+
+        /** Whether timezone is automatically set */
+        automatic: PropTypes.bool,
+    }).isRequired,
```

Comment:
Pardon my ignorance (still going through the thread), but the previous code had all checks for `details.timezone?` which meant it could've been null, but here we're making it `required`. Can you confirm if we decided to change this now?

Reply:
I noticed that `const shouldShowLocalTime = !ReportUtils.hasExpensifyEmails([details.login])` changed to `const shouldShowLocalTime = !ReportUtils.hasExpensifyEmails([details.login]) && details.timezone;`, so I guess that is why this is safe to be required now.

(just pointing it now because I missed it before)